src/applications/gi/containers/LandingPage.jsx

```diff
@@ -54,11 +63,27 @@ export class LandingPage extends React.Component {
   }
 
   render() {
+    const {
+      appGibctLandingPageShowPercent,
+      appGibctLandingPageShowVideo,
+    } = this.props.flags;
+
+    const ready = appGibctLandingPageShowPercent !== undefined;
```

Comment:
This isn't idiomatic. 

Capturing the current state of flags can take up to 100ms. Launch darkly suggests bootstrapping the data on the server side e.g. there are apis that we can hit to pull the current state of the flags during the build process. You can also prevent rendering of a feature and use the `on` ready event to track when the feature flags have been fetched client side.

Reply:
There's a definitely a big difference in functionality if you fetch these flags at build time vs client-side. You could potentially bootstrap the flag fetching in the `platform/startup` code, so that you delay loading the whole app while that's in progress. It seems pretty hard to try to delay just the parts that are flagged inside an individual app.

Reply:
Yeah- it's mainly an issue if the boot strap and live flags don't match- then you will see a flash of the wrong variant. I assume we'll come up with some recommendations that'll handle common use cases. 

I'm not sure we want live updating of the flags everywhere. e.g. we wouldn't want someone's schema / fields to change while they're filling out a form. There's also transforming data between two variants.

Reply:
Another possibility is only reading the flag on page load/component mount so it doesn't interrupt the user's interaction with the current page.